### DISCUSSION TITLE
How to get an `IReflectable` of a zserio Object?

### DISCUSSION THREAD

**Discussion by johannes-wolf** [ndsev/zserio#515] [Q&A] [answered]
How can one retrieve an `IReflectablePtr` to an already existing object?
The goal is to deserialize blobs and pass them as reflectable. I could not find anything related to reading reflectables in the code/tests.

**Comment by mikir** [comment, marked as answer]
There should be a public method `reflectable()` on the generated zserio object. If not, just please make sure that the zserio object has been generated by using command line options `-withTypeInfoCode` and `-withReflectionCode`.

It could be helpfull to have a look to our [reflectable test](https://github.com/ndsev/zserio/blob/master/test/arguments/with_reflection_code/cpp/WithReflectionCodeTest.cpp) to see some examples how to use C++ reflectable interface.

**Reply by johannes-wolf** [reply]
Thank you, it works. Maybe I forgot one of the flags the first time I tried.
